mods/force-thick-frames-plus.wh.cpp

```diff
@@ -0,0 +1,142 @@
+                if (GET_Y_LPARAM(lParam) > (pos.top + 30)) {
+                    return HTBORDER;
+                } else {
+                    return HTCAPTION;
+                }
+            case HTBOTTOMRIGHT:
+            case HTBOTTOM:
+            case HTBOTTOMLEFT:
+                return HTBORDER;
+            default:
+                return lr;
+        }
+    }
+
+    return DefSubclassProc(hWnd, uMsg, wParam, lParam);
+}
+
+// https://github.com/sandboxie-plus/Sandboxie/blob/294966c7d6e99cd153ede87ad09aa39ef29e34c3/Sandboxie/core/dll/Win32.c#L25
+using NtUserCreateWindowEx_t =
+    HWND(WINAPI*)(DWORD dwExStyle,
+                      PUNICODE_STRING UnsafeClassName,
+                      LPCWSTR VersionedClass,
+                      PUNICODE_STRING UnsafeWindowName,
+                      DWORD dwStyle,
+                      LONG x,
+                      LONG y,
+                      LONG nWidth,
+                      LONG nHeight,
+                      HWND hWndParent,
+                      HMENU hMenu,
+                      HINSTANCE hInstance,
+                      LPVOID lpParam,
+                      DWORD dwShowMode,
+                      DWORD dwUnknown1,
+                      DWORD dwUnknown2,
+                      VOID* qwUnknown3);
+NtUserCreateWindowEx_t NtUserCreateWindowEx_Original;
+HWND WINAPI NtUserCreateWindowEx_Hook(DWORD dwExStyle,
+                                          PUNICODE_STRING UnsafeClassName,
+                                          LPCWSTR VersionedClass,
+                                          PUNICODE_STRING UnsafeWindowName,
+                                          DWORD dwStyle,
+                                          LONG x,
+                                          LONG y,
+                                          LONG nWidth,
+                                          LONG nHeight,
+                                          HWND hWndParent,
+                                          HMENU hMenu,
+                                          HINSTANCE hInstance,
+                                          LPVOID lpParam,
+                                          DWORD dwShowMode,
+                                          DWORD dwUnknown1,
+                                          DWORD dwUnknown2,
+                                          VOID* qwUnknown3) {
+    BOOL canMod = false;
+    if ((dwStyle & WS_CAPTION) == WS_CAPTION) {
+        if ((dwStyle & WS_THICKFRAME) == 0) {
+            dwStyle |= WS_THICKFRAME;
+            canMod = true;
+        }
+    }
+
+    HWND ret = NtUserCreateWindowEx_Original(
+        dwExStyle, UnsafeClassName, VersionedClass, UnsafeWindowName, dwStyle,
+        x, y, nWidth, nHeight, hWndParent, hMenu, hInstance, lpParam,
+        dwShowMode, dwUnknown1, dwUnknown2, qwUnknown3);
+    if (canMod && ret != NULL) {
+        WindhawkUtils::SetWindowSubclassFromAnyThread(ret, &antiResizeProc, 0);
```

Comment:
You don't remove the subclass on uninit, which probably means that processes that have the subclass set will just crash.

Reply:
Whoops. And you're right. I'm just not sure how to go about fixing that.